mesos-rxjava-example/mesos-rxjava-example-framework/src/main/java/org/apache/mesos/rx/java/example/framework/sleepy/State.java

```diff
@@ -6,18 +6,30 @@
 import org.slf4j.LoggerFactory;
 
 import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
 
 import static com.google.common.collect.Maps.newConcurrentMap;
 
 final class State {
     private static final Logger LOGGER = LoggerFactory.getLogger(State.class);
 
-    private Protos.FrameworkID fwId;
+    private final double cpusPerTask;
+    private final double memMbPerTask;
 
     @NotNull
     private final Map<Protos.TaskID, Protos.TaskState> taskStates;
+    @NotNull
+    private final AtomicInteger offerCounter = new AtomicInteger();
+    @NotNull
+    private final AtomicInteger totalTaskCounter = new AtomicInteger();
```

Comment:
After thinking about these fields some more, do they even need to be `AtomicInteger`s? Isn't the whole point of the Rx stuff that the operations will occur sequentially? Maybe I'm missing something...

Even if we do need them to be `AtomicInteger`s, there's no need to `synchronize` their getters, because the field values themselves aren't changing (they're `final`).